clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2882,11 +2946,12 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     // (of either the lambda or the function object).
     CXXRecordDecl *KernelObj =
         GetSYCLKernelObjectType(KernelCallerFunc)->getAsCXXRecordDecl();
+
+    KernelCallOperatorVisitor KernelCallOperator(KernelCallerFunc, KernelObj);
```

Comment:
Technically the isLambda check can be done here as well. You don't really need to construct this object for lambdas.

Reply:
I'm not sure I understand how this suggestion was applied, we still construct the object, but there is also an additional check on Lambda...